nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/HandleHttpRequest.java

```diff
@@ -537,10 +542,22 @@ public void shutdown() throws Exception {
             server.stop();
             server.destroy();
             server.join();
+            clearInit();
             getLogger().info("Shut down {}", new Object[]{server});
         }
     }
 
+    @OnPrimaryNodeStateChange
+    public void onPrimaryNodeChange(final PrimaryNodeState newState) {
+        if (runOnPrimary.get() && newState.equals(PrimaryNodeState.PRIMARY_NODE_REVOKED)) {
+            try {
+                shutdown();
+            } catch (final Exception shutdownException) {
+                getLogger().debug("Failed to shutdown following the Primary Node role revoked: " + shutdownException);
```

Comment:
I think in this situation, we can make this a WARN level message.  If it fails to shutdown, and this node becomes the master again in the future, it might fail to start.

Reply:
Also, small change suggestion, "Processor is configured to run only on Primary, but failed to shutdown HTTP server following revocation of primary node status."

Reply:
Thanks for your suggestion
I have change the log level and keep the exception info